wurst/event/DamageModify.wurst

```diff
@@ -0,0 +1,327 @@
+		constant int LEECH_DAMAGE = compiletime(GetRandomInt(100, 200)) // just for lulz
+
+		// Implementation #1
+		function onLeechEffect1()
+			let d = GetTriggerUnit().returnDamage(GetSpellTargetUnit(), LEECH_DAMAGE, true, 0)
+
+			d.source.addHP(d.amount) // Leech effect implementation #1
+			d.source.addEffect(Abilities.vampiricAuraTarget, "origin").destr()
+
+			d.releaseInstance()
+
+		// Implementation #2
+		function onLeechEffect2()
+			GetTriggerUnit().damage(GetSpellTargetUnit(), LEECH_DAMAGE, LEECH_ID)
+		
+		init
+			addDamageListener() d ->
+				if d.id == LEECH_ID	
+					d.source.addHP(d.amount) // Leech effect implementation #2
+					d.source.addEffect(Abilities.vampiricAuraTarget, "origin").destr()
+
+	// Damage resistance buff
+		constant BUFF_DURATION = 5.0
+
+		function onBuffEffect()
+			let target = GetSpellTargetUnit()
+			let effect = target.addEffect(Abilities.voodooAuraTarget, "origin")
+			ForDamageCallback cb = () d ->
+				if target == d.target
+					d.amount -= d.initial * .5
+			addDamageModifier(cb)
+			doAfter(BUFF_DURATION) ->
+				removeDamageModifier(cb)
+				effect.destr()
+
+*/
+package DamageModify
+import AbilityObjEditing
+import DamageDetection
+import OnUnitEnterLeave
+import ObjectIdGenerator
+import LinkedList
+import ClosureTimers
+
+@configurable public constant DAMAGE_ID_NONTRIGGERED = -1
+
+public interface ForDamageCallback
+	function run(Damage d)
+
+let damageModifiers = new LinkedList<ForDamageCallback>
+public function addDamageModifier(ForDamageCallback cb)
+	damageModifiers.add(cb)
+public function removeDamageModifier(ForDamageCallback cb)
+	damageModifiers.remove(cb)
+
+let damageListeners = new LinkedList<ForDamageCallback>
+public function addDamageListener(ForDamageCallback cb)
+	damageListeners.add(cb)
+public function removeDamageListener(ForDamageCallback cb)
+	damageListeners.remove(cb)
+
+/** Deals triggered damage from unit to target.
+	You can specify id if there's any unique id interactions.
+	Damage instance is saved after being destroyed. */
+public function unit.returnDamage(unit target, real amount, bool spell, int id) returns Damage
+	return new Damage(this, target, amount, spell, id)..saveInstance()..modify()..deal()
+
+/** Deals triggered damage from unit to target.
+	You can specify id if there's any unique id interactions. */
+public function unit.damage(unit target, real amount, bool spell, int id)
+	new Damage(this, target, amount, spell, id)..modify()..deal()
+
+/** Deals triggered damage from unit to target. */
+public function unit.damage(unit target, real amount, bool spell)
+	new Damage(this, target, amount, spell, 0)..modify()..deal()
+/** Deals triggered spell damage from unit to target. */
+public function unit.damage(unit target, real amount)
+	new Damage(this, target, amount, true, 0)..modify()..deal()
+
+// ConvertAttackType(7) is a hidden attack type with some unique properties. For
+// more information see goo.gl/9k8tn .
```

Comment:
Can you add a caveat "Note that the values in the table at this link are outdated, but the principle still applies."?